src/server/lock_manager/waiter_manager.rs

```diff
@@ -270,10 +270,10 @@ impl Waiter {
         let mut lock_info = self.wait_info.lock_info.clone();
         lock_info.set_duration_to_last_update_ms(
             self.last_updated_time
-                .map(|t| t.elapsed().as_millis() as u64)
+                // round up, so that duration in (0, 1ms] won't be treated as 0.
+                .map(|t| (t.elapsed().as_micros() as f64 / 1000.0).ceil() as u64)
```

Comment:
In my imagination the implementation may like `if millis == 0 { millis = 1 }` in this closure. This looks good enough too to me though (is it possible that it `elapsed` returns exactly 0?